### ISSUE TITLE
ShelterTonight: At-a-glance shelter bed availability info

### ISSUE THREAD

**Issue by fureigh** [codeforamerica/project-ideas#66]
**One-sentence summary:**
ShelterTonight is a simple tool to show which shelters have beds available tonight, with relevant eligibility filters and contact info.

**Project needs:**
- First: someone who can provide information about how shelters currently track and communicate their bed availability. (Presumably this varies from place to place.) This will inform feasibility.
- Then: all the things.
- Buy-in from at least one organization that wants to use it.

**Status:**
<img src="http://media.giphy.com/media/CQy36RiuPRhF6/giphy.gif" height="200" alt="Pie" />

---

**Why?**
In the absence of such a tool, shelter seekers either get referred to shelters that don't actually have room for them and people who might otherwise make referrals hesitate to do so. This use case has come up several times in conversations about social service referrals.

**Extended mix possibilities:**
- Provide a way to close the referral feedback loop.
- Incorporate this into any social service referral apps (OpenReferral-related, etc.).

**Comment by greggish** [comment]
there's some sort of shelter hotline program at SF 311...

mathias gibson may know more about it

**Comment by fureigh** [comment]
Thanks, @greggish. I could've sworn I heard someone say there's a place to see this info for San Francisco, but the online resources I found all just say to call 311. I'll try Mathias and calling 311 to ask them how that works.

**Comment by lippytak** [comment]
SF HSA uses a homegrown shelter reservation system called CHANGES (acronym
for something...). 311 and the designated shelter reservation sites use
that. The info isn't public as far as I know (confirmed as of ~1 yr
ago)...but hopefully I'm wrong!

**Comment by daguar** [comment]
CCing @mathias-gibson here — Matthias, can you shed light on whether shelter space info is available?

**Comment by mathias-gibson** [comment]
So SFHSA engaged SF311 on a project for 90 day shelter beds and it launched about a year ago, so there should be a decent log of data to look at.  They were trying to make shelter beds reservation process more equitable and accessible by having one central waitlist for all of the shelters, so someone would call in to put their name on the waitlist, then get an sms message via Twilio when a bed was ready.  I can put you in touch with the Deputy Director who managed the project over there for an up to date status on the project, Im sure he'd love to sit down and chat (Andy Maimoni).  So yeah, that info is totally available and centralized at 311

**Comment by daguar** [comment]
Looks like the waitlist itself may even be online:
http://www.sf311.org/index.aspx?page=809

**Comment by mathias-gibson** [comment]
bingo, should have included that on my reply

**Comment by fureigh** [comment]
Neat, thanks! @mathias-gibson, does that waitlist mean that the answer to "are there any unclaimed shelter beds available tonight" is pretty much always "no?" (This may be a question for Andy Maimoni.)

**Comment by daguar** [comment]
@fureigh — @mathias-gibson can jump in if I'm wrong, but I think the
waitlist is for 90 days out, more long-term shelter bed stuff. The same-day
shelter availability seems to be all done in person.

On Mon, Mar 9, 2015 at 11:34 AM, Fureigh notifications@github.com wrote:

> Neat, thanks! @mathias-gibson https://github.com/mathias-gibson, does
> that waitlist mean that the answer to "are there any unclaimed shelter beds
> available tonight" is pretty much always "no?" (This may be a question for
> Andy Maimoni.)
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codeforamerica/project-ideas/issues/66#issuecomment-77914615
> .

## 

Dave Guarino
Engineer, Health
(2013 Fellowship alumnus)
Code for America http://www.codeforamerica.org/
dave@codeforamerica.org
LinkedIn http://www.linkedin.com/in/daveguarino/ | GitHub
https://github.com/daguar/

**Comment by RebeccaCoelius** [comment]
+1 @daguar

Its the "emergency" (<90 day) shelters that are relevant to this thread,
and most require you to line up in person early in the day, or get lucky
later in the day.

http://www.sf311.org/index.aspx?page=452

From 7am to 11pm a reservation can only be made by going in person to any
of the 7 city wide Single Adult Homeless Shelter Reservation Centers. Disabled
clients can make reservation by phone as an ADA accommodation
http://www.sf311.org/index.aspx?page=452#selectaresourcecenter.

Rebecca Coelius, MD
Director of Health
Code for America
415 298 2872
LinkedIn http://www.linkedin.com/in/rebeccacoelius
Twitter http://twitter.com/#!/RebeccaCoelius

**Comment by mathias-gibson** [comment]
Yup, Dave and Rebecca are both right. 311 only handles reservations for a
90 day bed right now (with plans to possibly add more services), shorter
reservations have to be made through CHANGES reservation sites and Homeless
Resource Centers.

Lots of great info is stored on this pdf for some reason:
http://www.sfhsa.org/asset/HousingHomelessServices/311_Waitlist_-_Combined_Document_-_5-1-14.pdf

**Comment by fureigh** [comment]
Thanks for the clarification, all! And @mathias-gibson, thanks for that PDF. And for the phrase "Lots of great info is stored on this pdf for some reason," which I'm now imagining framed at the CfA office.

I wish the reservation-by-phone system designed as an ADA accommodation could be leveraged with some Twilio transcription magic to yield the newest read-only phone-based API, but I strongly suspect it's a manual system. (The [link](http://www.sf311.org/index.aspx?page=452#selectaresourcecenter) bizarrely links to the same page, but it sounds like the "MAP Shelter Reservations Center" is the same system.)

Which begs the question, when a MAP reservation agent "go[es] offline to find an available shelter," how do they get information about shelter availability? Are they consulting an online system, calling around, or...?

**Comment by mathias-gibson** [comment]
Twilio is utilized by SF311 for notification of people on the 90 day
shelter waitlist, but thanks to procurement hassles its coming off Andy
Maimoni's (Deputy Director) personal credit card.  Procurement!

**Comment by RebeccaCoelius** [comment]
@daguar want to be sure you see the Twilio bit above

On Tuesday, March 10, 2015, Mathias Gibson notifications@github.com wrote:

> Twilio is utilized by SF311 for notification of people on the 90 day
> shelter waitlist, but thanks to procurement hassles its coming off Andy
> Maimoni's (Deputy Director) personal credit card. Procurement!
> 
> On Mon, Mar 9, 2015 at 11:50 PM, Fureigh <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
> 
> > Thanks for the clarification, all! And @mathias-gibson
> > https://github.com/mathias-gibson, thanks for that PDF. And for the
> > phrase "Lots of great info is stored on this pdf for some reason," which
> > I'm now imagining framed at the CfA office.
> > 
> > I wish the reservation-by-phone system designed as an ADA accommodation
> > could be leveraged with some Twilio transcription magic to yield the
> > newest
> > read-only phone-based API, but I strongly suspect it's a manual system.
> > (The link <
> > http://www.sf311.org/index.aspx?page=452#selectaresourcecenter>
> > bizarrely links to the same page, but it sounds like the "MAP Shelter
> > Reservations Center" is the same system.)
> > 
> > Which begs the question, when a MAP reservation agent "go[es] offline to
> > find an available shelter," how do they get information about shelter
> > availability? Are they consulting an online system, calling around,
> > or...?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <
> > https://github.com/codeforamerica/project-ideas/issues/66#issuecomment-78003988
> > 
> > .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/codeforamerica/project-ideas/issues/66#issuecomment-78156096
> .

## 

Rebecca Coelius, MD
Director of Health
Code for America
415 298 2872
LinkedIn http://www.linkedin.com/in/rebeccacoelius
Twitter http://twitter.com/#!/RebeccaCoelius

**Comment by fureigh** [comment]
Eesh, procurement.

Unless someone has a personal connection, I think I'll call a MAP reservation agent and try to chat with them.

**Comment by rcackermanCC** [comment]
@lippytak Would it be worth it to have @fureigh talk to Bevan Dufty? He'd probably know a fair number of people at the shelters. (Apologies if y'all are already talking to him.)

**Comment by fureigh** [comment]
@jamiealbrecht, you mentioned someone reaching out to you with a similar need. Do you remember who that was?
